In [1]:
import pandas as pd


In [3]:
file_path = '../data/raw/VDP006_01.csv'

In [7]:
df = pd.read_csv(file_path, sep=';', encoding='utf-8-sig')

As we are working with a dataset from an spanish org. (MIVAU), we set `sep=';'` because commas are used for decimals in Spain. 

In [ ]:
df.head(5) #A view of the first 5 rows of the dataset

,Año,Trimestre,Valor,Régimen,CPRO,Provincia,Comunidad_Autónoma,CODAUTO
0,1995,1,NaN,Libre,NaN,Ceuta y Melilla,Ceuta y Melilla,0.0
1,1995,2,NaN,Libre,NaN,Ceuta y Melilla,Ceuta y Melilla,0.0
2,1995,3,NaN,Libre,NaN,Ceuta y Melilla,Ceuta y Melilla,0.0
3,1995,4,NaN,Libre,NaN,Ceuta y Melilla,Ceuta y Melilla,0.0
4,1996,1,NaN,Libre,NaN,Ceuta y Melilla,Ceuta y Melilla,0.0


In [9]:
df.shape #The shape of the dataset (rows, columns)

(13711, 8)

In [10]:
df.columns #A view of the column names in the dataset

Index(['Año', 'Trimestre', 'Valor', 'Régimen', 'CPRO', 'Provincia',
       'Comunidad_Autónoma', 'CODAUTO'],
      dtype='str')

In [11]:
df.info() #A summary of the dataset, including data types and non-null values

<class 'pandas.DataFrame'>
RangeIndex: 13711 entries, 0 to 13710
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Año                 13711 non-null  int64  
 1   Trimestre           13711 non-null  int64  
 2   Valor               12867 non-null  float64
 3   Régimen             13711 non-null  str    
 4   CPRO                10176 non-null  float64
 5   Provincia           13711 non-null  str    
 6   Comunidad_Autónoma  13710 non-null  str    
 7   CODAUTO             13286 non-null  float64
dtypes: float64(3), int64(2), str(3)
memory usage: 857.1 KB


Fortunately, Python reads the parameters of the column named *'Valor'* as *float64*, not as objects, so we can use them in order to calculate statistical operations.

In [12]:
df.isna().sum() #Count of missing values in each column

Año                      0
Trimestre                0
Valor                  844
Régimen                  0
CPRO                  3535
Provincia                0
Comunidad_Autónoma       1
CODAUTO                425
dtype: int64

## Data Diagnosis

### Matrix Dimension
We are working with 13,711 observations x 8 variables.

### Data Types
Data types are correctly assigned for almost every variable:
- `Año` and `Trimestre` are `int64`, which is ideal for future exploratory analysis.
- `Valor` is `float64`, allowing us to apply descriptive statistics.
- `Régimen`, `Provincia`, and `Comunidad_Autónoma` are `str`.
- `CPRO` and `CODAUTO` are `float64` because data is missing in some rows, forcing pandas to convert the whole column to `float64` (Python integers cannot hold null values natively).

### Missing Data
- `Valor` (844 nulls - 6.1%): There are 844 quarters/zones where the administration did not record prices (or they were not published).
- `CPRO` (3,535 nulls - 25.7%): About 25% of rows do not have a provincial code. This may not be an error; in public datasets, official sources often include summary rows (e.g., 'Total España' or 'Total Andalucía') that do not have a unique province assigned.
- `Comunidad_Autónoma` (1 null): This is the only row in the dataset missing an assigned community. We should investigate the reason behind this.